### ISSUE TITLE
LANDSAT8 collection downloading not working

### ISSUE THREAD

**Issue by JDESLOIRES** [sentinel-hub/sentinelhub-py#209]
**Describe the bug**

I cannot have images from LANDSAT8 data collections. I tried using DataCollection.LANBDSAT8, DataCollection.LANBDSAT8_L1, DataCollection.LANBDSAT8_L2 but I still have very strange outputs (all pixels = 0).

**To Reproduce**

```python
import datetime as datetime
from eolearn.io import SentinelHubInputTask
from sentinelhub import BBox, CRS, DataCollection, MimeType, SentinelHubRequest, SHConfig
from eolearn.core import SaveTask, LoadTask, FeatureType, LinearWorkflow

bbox = [-0.966088771820068, 46.1229628297395, -0.961373448371887 ,46.1255691897157]
field_bbox = BBox(bbox=[(bbox[0], bbox[1]), (bbox[2], bbox[3])], crs=CRS.WGS84)

time_period = ('2018-10-15', '2019-11-01')
time_difference = datetime.timedelta(hours=2)

input_task = SentinelHubInputTask(
    data_collection=DataCollection.LANDSAT8_L1,
    bands=['B01', 'B02', 'B03', 'B04', 'B05', 'B06', 'B07'],
    bands_feature=(FeatureType.DATA, 'BANDS-L8'),
    additional_data=[(FeatureType.MASK, 'dataMask', 'IS_DATA')],
    resolution=30,
    maxcc=1,
    time_difference=time_difference,
    config=config,
    max_threads=3
)

workflow = LinearWorkflow(input_task)  
result_L8 = workflow.execute({
    input_task: {'bbox': field_bbox, 'time_interval': time_period}
})
```
**Expected behavior**

Normally I should have reflectance values but all pixels are equal to zero.

**Environment**

python 3.7 with sentinelhub==3.4.0 and eo-learn===0.9.2

**Comment by zigaLuksic** [comment]
Hi @JDESLOIRES

The bug you found comes from recent changes on the Sentinel Hub Service, where LANDSAT8 collections switched from digital numbers to reflectance. We are preparing a fix in the upcoming eo-learn release, but for the time being you can use `SentinelHubEvalscriptTask`.

Here is your script, but using (a correct) evalscript:
```python
import datetime as datetime
import numpy as np
from eolearn.io import SentinelHubEvalscriptTask
from sentinelhub import BBox, CRS, DataCollection, MimeType, SentinelHubRequest, SHConfig
from eolearn.core import SaveTask, LoadTask, FeatureType, LinearWorkflow


bbox = [-0.966088771820068, 46.1229628297395, -0.961373448371887 ,46.1255691897157]
field_bbox = BBox(bbox=[(bbox[0], bbox[1]), (bbox[2], bbox[3])], crs=CRS.WGS84)

time_period = ('2018-10-15', '2019-11-01')
time_difference = datetime.timedelta(hours=2)

evalscript = """
    function setup() {
        return {
            input: [{
                bands: ["B01", "B02", "B03", "B04", "B05", "B06", "B07", "dataMask"],
                units: ["REFLECTANCE", "REFLECTANCE", "REFLECTANCE", "REFLECTANCE", "REFLECTANCE", "REFLECTANCE", "REFLECTANCE", "DN"]
            }],
            output: [
                { id:"BANDS", bands:7, sampleType: SampleType.FLOAT32 }, { id:"IS_DATA", bands:1, sampleType: SampleType.UINT8 }
            ]
        }
    }

    function evaluatePixel(sample) {
        return {BANDS: [sample.B01, sample.B02, sample.B03, sample.B04, sample.B05, sample.B06, sample.B07], IS_DATA: [sample.dataMask]};
    }
"""


input_task = SentinelHubEvalscriptTask(
    features=[(FeatureType.DATA, 'BANDS', 'BANDS-L8'), (FeatureType.MASK, 'IS_DATA')],
    data_collection=DataCollection.LANDSAT8_L1,
    evalscript=evalscript,
    resolution=30,
    maxcc=1,
    time_difference=time_difference,
    max_threads=3
)

workflow = LinearWorkflow(input_task)
result_L8 = workflow.execute({
    input_task: {'bbox': field_bbox, 'time_interval': time_period}
})
```

**Comment by JDESLOIRES** [comment]
Hi @zigaLuksic ,

Many thanks for your fast reply!

Just one last question : For Landsat 8 **cloud masking**, I was using the band 'BQA' for L1C preprocessing level (values 3 and 4) but it seems to not be available for L2A ... How can I access to it?

Best regards,
Johann

**Comment by zigaLuksic** [comment]
Hi @JDESLOIRES 

according to the [documentation](https://docs.sentinel-hub.com/api/latest/data/landsat-8-l2/#available-bands-and-data) it should be available.  Do you perhaps have an example where this issue arises?

**Comment by JDESLOIRES** [comment]
My mistake, I had to call  decodeL8C2Qa(sample.BQA).cloud in the evalscript to get the cloud binary mask directly, instead of using the values 3 and 4 as previously with Landsat8_L1C (it was working like this last year).

Thanks again!

**Comment by AleksMat** [comment]
Closing the issue as this should be solved with `eo-learn==0.10.1` and `sentinelhub==3.4.1`.
